plutus-ledger-api/src/PlutusLedgerApi/V3/Contexts.hs

```diff
@@ -0,0 +1,625 @@
+{-# LANGUAGE DeriveAnyClass    #-}
+{-# LANGUAGE DerivingVia       #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards   #-}
+{-# LANGUAGE TemplateHaskell   #-}
+{-# OPTIONS_GHC -Wno-simplifiable-class-constraints #-}
+{-# OPTIONS_GHC -fno-omit-interface-pragmas #-}
+{-# OPTIONS_GHC -fno-specialise #-}
+{-# OPTIONS_GHC -fno-strictness #-}
+
+module PlutusLedgerApi.V3.Contexts where
+
+import Data.String
+import GHC.Generics (Generic)
+import Prettyprinter (nest, vsep, (<+>))
+import Prettyprinter.Extras
+
+import PlutusLedgerApi.V2 qualified as V2
+import PlutusTx qualified
+import PlutusTx.AssocMap hiding (filter, mapMaybe)
+import PlutusTx.Prelude qualified as PlutusTx
+import PlutusTx.Show qualified as PlutusTx
+
+import Prelude qualified as Haskell
+
+newtype ColdCommitteeCredential = ColdCommitteeCredential V2.Credential
+  deriving stock (Generic)
+  deriving (Pretty) via (PrettyShow ColdCommitteeCredential)
+  deriving newtype
+    ( Haskell.Eq
+    , Haskell.Show
+    , PlutusTx.Eq
+    , PlutusTx.ToData
+    , PlutusTx.FromData
+    , PlutusTx.UnsafeFromData
+    )
+
+newtype HotCommitteeCredential = HotCommitteeCredential V2.Credential
+  deriving stock (Generic)
+  deriving (Pretty) via (PrettyShow HotCommitteeCredential)
+  deriving newtype
+    ( Haskell.Eq
+    , Haskell.Show
+    , PlutusTx.Eq
+    , PlutusTx.ToData
+    , PlutusTx.FromData
+    , PlutusTx.UnsafeFromData
+    )
+
+newtype DRepCredential = DRepCredential V2.Credential
+  deriving stock (Generic)
+  deriving (Pretty) via (PrettyShow DRepCredential)
+  deriving newtype
+    ( Haskell.Eq
+    , Haskell.Show
+    , PlutusTx.Eq
+    , PlutusTx.ToData
+    , PlutusTx.FromData
+    , PlutusTx.UnsafeFromData
+    )
+
+data DRep
+  = DRep DRepCredential
```

Comment:
Since there's already a `DRepCredential`, I renamed this to `DRep`. @lehins @WhatisRT Let me know if there's a better name!

Reply:
Yeah,I think `DRep` is a perfect name for this type. That's what we call it in ledger too.